code/modules/atmospherics/machinery/components/unary_devices/vent_pump.dm

```diff
@@ -42,16 +54,40 @@
 			tool_screentips = string_assoc_nested_list(list(
 				TOOL_MULTITOOL = list(
 					SCREENTIP_CONTEXT_LMB = "Log to link later with air sensor",
-				)
+				),
+				TOOL_SCREWDRIVER = list(
+					SCREENTIP_CONTEXT_LMB = "Repair",
+				),
 			))
 		AddElement(/datum/element/contextual_screentip_tools, tool_screentips)
 	. = ..()
+	sound_loop = new(src)
 	assign_to_area()
 
 /obj/machinery/atmospherics/components/unary/vent_pump/examine(mob/user)
 	. = ..()
 	. += span_notice("You can link it with an air sensor using a multitool.")
 
+	if(fan_overclocked)
+		. += span_warning("It is currently overclocked causing it to take damage over time.")
+
+	if(fan_integrity > 0)
```

Comment:
Then after using the normal integrity system, you can store and update the condition string as integrity is updated instead of remaking it each time in examine